dask/bag/text.py

```diff
@@ -61,13 +64,22 @@ def read_text(urlpath, blocksize=None, compression='infer',
     --------
     from_sequence: Build bag from Python sequence
     """
+    files = open_files(urlpath, mode='rt', encoding=encoding,
+                       errors=errors, compression=compression,
+                       **(storage_options or {}))
     if blocksize is None:
-        files = open_files(urlpath, mode='rt', encoding=encoding,
-                           errors=errors, compression=compression,
-                           **(storage_options or {}))
-        blocks = [delayed(list)(delayed(file_to_blocks)(fil)) for fil in files]
+        if files_per_partition is None:
+            blocks = [delayed(list)(delayed(file_to_blocks)(fil)) for fil in files]
+        else:
+            blocks = []
+            for start in range(0, len(files), files_per_partition):
+                block_files = files[start:(start + files_per_partition)]
+                blocks.append(delayed(list)(delayed(files_to_blocks)(block_files)))
 
     else:
+        if files_per_partition is not None:
+            raise ValueError('Only one of blocksize or files_per_partition can be set')
```

Comment:
I find it slightly easier to read if this goes to the top of this function in a snippet like the following:

```python
if blocksize and files_per_partition:
    raise ValueError("...")
```

I think that navigating the nested if/else tree requires a bit more mental effort from future readers.  I think that readers are also accustomed to seeing validation code at the tops of functions.